src/classes/TDTM_Runnable_TEST.cls

```diff
@@ -36,65 +36,132 @@
-        acc = [select Name from Account where Id = :acc.Id];
+        Contact actualContact = [SELECT FirstName FROM Contact WHERE Id = :contact.Id];
+        Account actualAccount = [SELECT Name FROM Account WHERE Id = :acc.Id];
         
-        System.assertEquals('New First Name', contact.FirstName); //To verify fields from record are available in future
-        System.assertEquals('New Acc Name', acc.Name);
+        // Ensure 'ERR_ParentAccountUpdater2_TEST' trigger handler future method processing has been applied
+        System.assertEquals(runFutureContactName, actualContact.FirstName); 
+        System.assertEquals(runFutureAccountName, actualAccount.Name);
     }
-    
+
     /*************************************************************************** 
-    * @description Test we do not get the "Cannot have more than 10 chunks in a single operation. Please rearrange the data to 
-    * reduce chunking" error. 
-    * If you comment out the first line in the method groupSObjects of TDTM_Runnable this test fails.
+    * @description Verifies duplicate records are removed before DML operations
+    * in order to prevent "Duplicate id in list" error.
     **/
-    public static testmethod void nochunking() {
-    	if (strTestOnly != '*' && strTestOnly != 'nochunking') return;
-    	
-    	npe01__Contacts_and_Orgs_Settings__c contactSettingsForTests = UTIL_CustomSettingsFacade.getContactsSettingsForTests(
-            new npe01__Contacts_and_Orgs_Settings__c (npe01__Account_Processor__c = CAO_Constants.HH_ACCOUNT_PROCESSOR));
+    @isTest
+    private static void groupByTypeShouldRemoveDuplicateSObjects() {
+        List<Id> accountIds = extractIds([SELECT Id FROM Account]);
+        List<Id> contactIds = extractIds([SELECT Id FROM Contact]);
+
+        List<SObject> objectsToInsert = new List<SObject>{
+            new Contact(LastName = 'GroupBySmith', AccountId = accountIds[0]),
+            new Contact(LastName = 'GroupByClark', AccountId = accountIds[0]),
+            new Contact(LastName = 'GroupByClark', AccountId = accountIds[1]),
+            new Contact(LastName = 'GroupByClark', AccountId = accountIds[1])
+        };
+
+        List<SObject> objectsToUpdate = new List<SObject>{
+            new Contact(Id = contactIds[0], LastName = 'GroupBySmith', AccountId = accountIds[0]),
+            new Contact(Id = contactIds[1], LastName = 'GroupByClark', AccountId = accountIds[1]),
+            new Contact(Id = contactIds[0], LastName = 'GroupBySmith', AccountId = accountIds[0]),
+            new Contact(Id = contactIds[2], LastName = 'GroupByClark', AccountId = accountIds[1])
+        };
+
+        TDTM_Runnable.DmlWrapper dmlWrapper = new TDTM_Runnable.DmlWrapper();
+        dmlWrapper.objectsToInsert.addAll(objectsToInsert);
+        dmlWrapper.objectsToUpdate.addAll(objectsToUpdate);
+
+        dmlWrapper.groupByType();
+
+        System.assertEquals(3, dmlWrapper.objectsToInsert.size(),
```

Comment:
That is the question I wanted to ask, ie should we remove SObjects on insert that are the same. The code has been implemented a long time. Is this expected behaviour?